criu/memfd.c

```diff
@@ -167,7 +166,11 @@ static int dump_one_memfd(int lfd, u32 id, const struct fd_parms *p)
 		link = p->link;
 
 	strip_deleted(link);
-	name = &link->name[1+MEMFD_PREFIX_LEN];
+	/* link->name is always started with "." what has to be skipped.  */
+	if (strncmp(link->name + 1, MEMFD_PREFIX, MEMFD_PREFIX_LEN) == 0)
+		name = &link->name[1 + MEMFD_PREFIX_LEN];
+	else
+		name = link->name + 1;
```

Comment:
lrw------- 1 root root 64 Mar 18 08:25 f7f5f000-f7f60000 -> '/dev/zero (deleted)'

Reply:
So that's a bit strange.

`/dev/zero` is some sort of kernel placeholder to show something, but semantically, we don't have a name. We could:
1) Make `name` optional in the memfd image format. But then, what should we use during restore?
2) Or we can use the name `"anonymous"` (instead of `link->name+1`), which is a reasonable name to use when restoring the region.

Reply:
Actually, in the kernel source, this is a real name:
https://github.com/torvalds/linux/blob/master/mm/shmem.c#L4176

I mean they could use any name there but they chose /dev/zero. I don't think that we need to reinvent a file name for anon mappings...